### PR TITLE
Gui list

### PR DESCRIPTION
**Pull request by markhelmus**

### CODE REVIEW

**Comment by FrankTN** [comment]
You should probably remove the .idea files and update the .gitignore to prevent the project from getting messy

File: Hestia/app/src/main/java/com/rugged/application/hestia/Client.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    public void sendActionRequest(Integer id, String action) {
+        // Write JSON
+        JSONObject jsonObj = new JSONObject();
+        // Stripping the text from the new-line separators is necessary to avoid
+        // Strings line "abc...\n" -> the "\n" would be visible
+        jsonObj.put("id", id);
+        jsonObj.put("action", action);
+        this.writeJSONToServer(jsonObj);
```
Comment (FrankTN) [inline Hestia/app/src/main/java/com/rugged/application/hestia/Client.java]: This should comply with the JSON object as defined on basecamp, so:
`
{
  "action": {
    "id": 1,
    "type": "Lock"
  }
}
`

**Review by FrankTN — CHANGES_REQUESTED** [review]
In general, this looks good. There are a lot of hardcoded things which need to change at a later stage.

My main concern is with the JSON specification, which does not agree completely with the one published on Basecamp.

I also recommend adding comments in critical sections, to increase readability.

File: Hestia/app/src/main/java/com/rugged/application/hestia/PeripheralFragment.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            @Override
+            public void onClick(View view) {
+                new SendJSONFile().execute("82.73.173.179", "openLock");
+            }
+        });
+
+        offButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                new SendJSONFile2().execute("82.73.173.179", "openLock");
```
Comment (suzannemaquelin) [inline Hestia/app/src/main/java/com/rugged/application/hestia/PeripheralFragment.java]: For the off button, shouldn't it be "closeLock"?

File: Hestia/app/src/main/java/com/rugged/application/hestia/PeripheralLab.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
+package com.rugged.application.hestia;
+
+import android.content.Context;
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class PeripheralLab {
```
Comment (suzannemaquelin) [inline Hestia/app/src/main/java/com/rugged/application/hestia/PeripheralLab.java]: It's probably just me, but I don't understand what peripheralLab and sPeripheralLab are

**Review by suzannemaquelin — CHANGES_REQUESTED** [review]
I think you made a mistake with "openLock". For the off button it should be closeLock

**Review by Feikoritsema — APPROVED** [review]
It seems weird to review my own code, I read through it once more and approve.

**Review by FrankTN — APPROVED** [review]

File: Hestia/app/src/main/java/com/rugged/application/hestia/PeripheralLab.java (hunk 3 above)
Comment (FrankTN) [reply Hestia/app/src/main/java/com/rugged/application/hestia/PeripheralLab.java]: I had the same thing, its probably a good idea to add some comments in these parts.

Comment (markhelmus) [reply Hestia/app/src/main/java/com/rugged/application/hestia/PeripheralLab.java]: I added some explaining comments. Hopefully it is understandable what those classes are doing. If not, please let me know.

**Review by suzannemaquelin — APPROVED** [review]

**Pull request merged**